### ISSUE TITLE
plugins.crunchyroll: Plugin fails on beta URLs

### ISSUE THREAD

**Issue by orbitalflower** [streamlink/streamlink#4790]
### Checklist

- [X] This is a plugin issue and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed plugin issues](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22plugin+issue%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

Latest build from the master branch

### Description

Crunchyroll plugin throws a ValidationError on Crunchyroll Beta URLs.

Non-beta URLs still seem to work, but Crunchyroll automatically serves beta URLs to logged-in users.

Error occurs on 4.3.0 as well as latest master.

### Debug log

```text
$ streamlink https://beta.crunchyroll.com/watch/G50UZ9GM4/wolf-and-best-clothes
[cli][debug] OS:         Linux-5.15.0-43-generic-x86_64-with-glibc2.35
[cli][debug] Python:     3.10.4
[cli][debug] Streamlink: 4.3.0+32.gf0202598
[cli][debug] Dependencies:
[cli][debug]  PySocks: 1.7.1
[cli][debug]  isodate: 0.6.1
[cli][debug]  lxml: 4.8.0
[cli][debug]  pycountry: 20.7.3
[cli][debug]  pycryptodome: 3.14.1
[cli][debug]  requests: 2.27.1
[cli][debug]  websocket-client: 1.2.3
[cli][debug]  importlib-metadata: 4.6.4
[cli][debug] Arguments:
[cli][debug]  url=https://beta.crunchyroll.com/watch/G50UZ9GM4/wolf-and-best-clothes
[cli][debug]  --loglevel=debug
[cli][debug]  --player=vlc
[cli][debug]  --default-stream=['best']
[cli][debug]  --crunchyroll-username=********
[cli][debug]  --crunchyroll-password=********
[cli][info] Found matching plugin crunchyroll for URL https://beta.crunchyroll.com/watch/G50UZ9GM4/wolf-and-best-clothes
error: Unable to validate response text: ValidationError(NoneOrAllSchema):
  ValidationError(NoneOrAllSchema):
    ValidationError(NoneOrAllSchema):
      ValidationError(dict):
        Unable to validate value of key 'content'
        Context(dict):
          Key 'byId' not found in <{'media': {'byId': {'G50UZ9GM4': {'id': 'G50UZ9GM4', 't...>
```

**Comment by gravyboat** [comment]
Duplicate of https://github.com/streamlink/streamlink/issues/3674

**Comment by bastimeyer** [comment]
Not a duplicate...

> [cli][debug] Streamlink: 4.3.0+32.gf0202598

----

> Key 'byId' not found in <{'media': {'byId': {'G50UZ9GM4': {'id': 'G50UZ9GM4', 't...>

This looks like a small change in the response JSON data

**Comment by gravyboat** [comment]
@bastimeyer I don't see how it's not a duplicate even though it's a different release version. No one ever worked to fix the crunchyroll plugin for the beta version of the site and we agreed we wouldn't work on the beta version of the site till they finalize the release. The fact the response validator output is different just goes to show there have been further changes with the beta.

**Comment by bastimeyer** [comment]
> I don't see how it's not a duplicate even though it's a different release version.
> there have been further changes with the beta

As you can see in the plugin's history/blame, the validation schema that was added in #4510 doesn't work anymore due to a recent change of the response JSON data. It closed #3674 because it was working at the time...
https://github.com/streamlink/streamlink/blame/f0202598322692a757e9660c9d952ce689f7d606/src/streamlink/plugins/crunchyroll.py#L306-L309

It's not a duplicate because the plugin needs fixing again. Whether it's caused by an unstable change due to the site being in beta doesn't really matter, even if it's annoying. The other thread was not a "crunchyroll-beta meta thread", so let's not close this one.

Btw, as said, the change looks rather simple and can probably be fixed quite easily. I won't take a look at this myself though, because I avoid content like this, if you know what I mean. No further comment. :)

**Comment by mkbloke** [comment]
It does not look like it is that easy to fix.  The method I was using before was to avoid the Cloudflare intermediate page that Streamlink could not get through.  It appears there is now no data linking to the non-beta URL from a beta URL.

Visiting a beta URL appears to redirect to the (language appropriate) home page.  It is still possible to remove `beta.` from the beta URL in a browser and get redirected to the non-beta URL, but as above, this won't work in Streamlink as it gets stuck on the intermediate Cloudflare page.
